crates/ibc/src/applications/transfer/acknowledgement.rs

```diff
@@ -31,68 +31,115 @@ pub enum Acknowledgement {
 }
 
-impl Acknowledgement {
+impl TokenTransferAcknowledgement {
     pub fn success() -> Self {
         Self::Success(ConstAckSuccess::Success)
     }
 
     pub fn from_error(err: TokenTransferError) -> Self {
         Self::Error(format!("{ACK_ERR_STR}: {err}"))
     }
+
+    pub fn is_successful(&self) -> bool {
+        matches!(self, TokenTransferAcknowledgement::Success(_))
+    }
 }
 
-impl AsRef<[u8]> for Acknowledgement {
-    fn as_ref(&self) -> &[u8] {
+impl Display for TokenTransferAcknowledgement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
         match self {
-            Acknowledgement::Success(_) => ACK_SUCCESS_B64.as_bytes(),
-            Acknowledgement::Error(s) => s.as_bytes(),
+            TokenTransferAcknowledgement::Success(_) => write!(f, "{ACK_SUCCESS_B64}"),
+            TokenTransferAcknowledgement::Error(err_str) => write!(f, "{err_str}"),
         }
     }
 }
 
-impl Display for Acknowledgement {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
-        match self {
-            Acknowledgement::Success(_) => write!(f, "{ACK_SUCCESS_B64}"),
-            Acknowledgement::Error(err_str) => write!(f, "{err_str}"),
+impl From<TokenTransferAcknowledgement> for Vec<u8> {
+    fn from(ack: TokenTransferAcknowledgement) -> Self {
+        match ack {
+            TokenTransferAcknowledgement::Success(_) => r#"{"result":"AQ=="}"#.as_bytes().into(),
+            // TokenTransferAcknowledgement::Error(s) => s.as_bytes(),
+            TokenTransferAcknowledgement::Error(s) => alloc::format!(r#"{{"error":"{s}"}}"#).into(),
         }
     }
 }
 
-impl AckTrait for Acknowledgement {}
+impl From<TokenTransferAcknowledgement> for Acknowledgement {
+    fn from(ack: TokenTransferAcknowledgement) -> Self {
+        ack.into()
+    }
+}
 
 #[cfg(test)]
 mod test {
     use super::*;
 
     #[test]
     fn test_ack_ser() {
-        fn ser_json_assert_eq(ack: Acknowledgement, json_str: &str) {
+        fn ser_json_assert_eq(ack: TokenTransferAcknowledgement, json_str: &str) {
             let ser = serde_json::to_string(&ack).unwrap();
             assert_eq!(ser, json_str)
         }
 
-        ser_json_assert_eq(Acknowledgement::success(), r#"{"result":"AQ=="}"#);
         ser_json_assert_eq(
-            Acknowledgement::Error("cannot unmarshal ICS-20 transfer packet data".to_owned()),
+            TokenTransferAcknowledgement::success(),
+            r#"{"result":"AQ=="}"#,
+        );
+        ser_json_assert_eq(
+            TokenTransferAcknowledgement::Error(
+                "cannot unmarshal ICS-20 transfer packet data".to_owned(),
+            ),
             r#"{"error":"cannot unmarshal ICS-20 transfer packet data"}"#,
         );
     }
 
+    #[test]
+    fn test_ack_success_to_vec() {
+        let ack_success = TokenTransferAcknowledgement::success();
+
+        // Check that it's the same output as ibc-go
+        assert_eq!(
+            Vec::<u8>::from(ack_success),
```

Comment:
See if this way is acceptable to you :)

```suggestion
        let ack_success: Vec<u8> = TokenTransferAcknowledgement::success().into();

        // Check that it's the same output as ibc-go
        assert_eq!(
            ack_success,
            r#"{"result":"AQ=="}"#.as_bytes()
        );
```

Reply:
[03b64c7](https://github.com/cosmos/ibc-rs/pull/364/commits/03b64c70d47cb06bd820bdfa1dd1fad0d5ebf9bf)